### PR TITLE
Feature/distribution client proxy

### PR DESCRIPTION
**Pull request by gpoleze**
# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: distribution-client-proxy/src/main/java/ca/bc/gov/ag/proxy/service/DocumentDistributionService.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import java.util.Base64;
+
+public class DocumentDistributionService {
+    private final String soapEndpointUrl;
+    private final String soapAction;
+    private final DocumentDistributionRequest documentDistributionRequest;
+    private final SOAPMessage soapMessage;
+    private final SOAPEnvelope soapEnvelope;
+
+    private final String documentDistributionRequestNamespace = "DocumentDistributionRequest";
```
Comment (KDNeufeld) [inline distribution-client-proxy/src/main/java/ca/bc/gov/ag/proxy/service/DocumentDistributionService.java:27]: unused variable

**Comment by KDNeufeld** [comment]
It's a good idea to usually include a **.dockerignore** file in your project folder next to the **.gitignore**.  Usually these have the same or similar content.  When building your app using docker, you want to exclude folders with prebuilt class files so docker builds them from scratch instead of copying them from your Eclipse workspace.  In this case, the file would include "/target".

File: distribution-client-proxy/src/main/java/ca/bc/gov/ag/proxy/service/DocumentDistributionService.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+        String userAndPassword = String.format("%s:%s", username, password);
+
+        String basicAuth = Base64.getEncoder().encodeToString(userAndPassword.getBytes());
+        headers.addHeader("Authorization", "Basic " + basicAuth);
+
+        soapMessage.saveChanges();
+
+        /* Print the request message, just for debugging purposes */
+        System.out.println("Request SOAP Message:");
```
Comment (KDNeufeld) [inline distribution-client-proxy/src/main/java/ca/bc/gov/ag/proxy/service/DocumentDistributionService.java:106]: Use logger instead of System.out

**Review by KDNeufeld — APPROVED** [review]
I've tested the main jag-efax application and it still works after these changes.  I could not find a way to test the proxy to make sure it works.  There is some minor code cleanup, but if you think this will work we can merge and try it out in TEST.

File: distribution-client-proxy/src/main/java/ca/bc/gov/ag/proxy/service/DocumentDistributionService.java (hunk 1 above)
Comment (gpoleze) [reply distribution-client-proxy/src/main/java/ca/bc/gov/ag/proxy/service/DocumentDistributionService.java]: The variable was deleted

**Pull request merged**